src/decider/agent/DeciderAgent.php

```diff
@@ -230,6 +230,10 @@ private function processItem(Item $item)
 
     $haveDecided = false;
 
+    if (($this->activeRules&self::RULES_OJO_NO_CONTRADICTION)== self::RULES_OJO_NO_CONTRADICTION) {
+      $haveDecided = $this->autodecideIfOjoMatchesNoContradiction($itemTreeBounds, $projectedScannerMatches);
+    }
+
     if (($this->activeRules&self::RULES_NOMOS_IN_MONK)== self::RULES_NOMOS_IN_MONK) {
```

Comment:
shouldn't the `autodecideNomosMatchesInsideMonk be prevented from overwriting the possible `true` from `autodecideIfOjoMatchesNoContradiction`, i.e.
```suggestion
    if (!$haveDecided && ($this->activeRules&self::RULES_NOMOS_IN_MONK)== self::RULES_NOMOS_IN_MONK) {
```